h/links.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+
+"""
+Provides links to different representations of annotations.
+"""
+
+
+from h._compat import urlparse
+
+
+def html_link(request, annotation):
+    """Generate a link to an HTML representation of an annotation."""
+    return request.route_url('annotation', id=annotation.id)
+
+
+def incontext_link(request, annotation):
+    """Generate a link to an annotation on the page where it was made."""
+    is_reply = bool(annotation.references)
+    if not request.feature('direct_linking') or is_reply:
+        return None
+
+    bouncer_url = request.registry.settings.get('h.bouncer_url')
+    if not bouncer_url:
+        return None
+    return urlparse.urljoin(bouncer_url, annotation.id)
```

Comment:
Yeah, I'm not sure I necessarily prefer this over using a named static route as the previous implementation did, but it's fine for now.